From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Date: Wed, 22 Feb 2017 17:31:51 -0800
Subject: [PATCH] Add an option to disable the curses module

Upstream: N/A

Signed-off-by: Thomas Petazzoni <thomas.petazzoni@free-electrons.com>
Signed-off-by: Samuel Martin <s.martin49@gmail.com>
[ Andrey Smirnov: ported to Python 3.6 ]
Signed-off-by: Andrey Smirnov <andrew.smirnov@gmail.com>
[ Adam Duskett: ported to Python 3.10.0 ]
Signed-off-by: Adam Duskett <aduskett@gmail.com>
[ Vincent Fazio: ported to Python 3.13.2 ]
Signed-off-by: Vincent Fazio <vfazio@gmail.com>
[ Bernd Kuhls: ported to Python 3.15.0]
Signed-off-by: Bernd Kuhls <bernd@kuhls.net>
---
 Makefile.pre.in | 5 ++++-
 configure.ac    | 7 +++++++
 2 files changed, 11 insertions(+), 1 deletion(-)

diff --git a/Makefile.pre.in b/Makefile.pre.in
index 7e9eef79b06..174164142c0 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -2623,7 +2623,6 @@ LIBSUBDIRS=	asyncio \
 		concurrent concurrent/futures concurrent/interpreters \
 		csv \
 		ctypes ctypes/macholib \
-		curses \
 		dbm \
 		email email/mime \
 		encodings \
@@ -2835,6 +2834,10 @@ LIBSUBDIRS += tkinter
 TESTSUBDIRS += test/test_tkinter test/test_ttk
 endif
 
+ifeq (@CURSES@,yes)
+LIBSUBDIRS += curses
+endif
+
 COMPILEALL_OPTS=-j0
 
 ifeq (@PYDOC@,yes)
diff --git a/configure.ac b/configure.ac
index be9428dc2cb..1f28a1de74d 100644
--- a/configure.ac
+++ b/configure.ac
@@ -4988,6 +4988,13 @@ AC_ARG_ENABLE(sqlite3,
 AS_IF([test "$SQLITE3" = "no"],
       [PY_STDLIB_MOD_SET_NA([_sqlite3])])
 
+AC_SUBST(CURSES)
+AC_ARG_ENABLE(curses,
+	AS_HELP_STRING([--disable-curses], [disable curses]),
+	[ CURSES="${enableval}" ], [ CURSES=yes ])
+AS_IF([test "$CURSES" = "no"],
+      [PY_STDLIB_MOD_SET_NA([_curses], [_curses_panel])])
+
 AC_SUBST(PYDOC)
 
 AC_ARG_ENABLE(pydoc,
-- 
2.47.3

